src/Microsoft.AspNetCore.Razor.Tools/CompilerHost.cs

```diff
@@ -26,13 +26,33 @@ public override ServerResponse Execute(ServerRequest request, CancellationToken
                     return new RejectedServerResponse();
                 }
 
+                var exitCode = 0;
+                var output = string.Empty;
                 var app = new Application(cancellationToken);
                 var commandArgs = parsed.args.ToArray();
 
-                var exitCode = app.Execute(commandArgs);
-                var output = app.Out.ToString() ?? string.Empty;
+                if (ServerLogger.IsLoggingEnabled)
+                {
+                    using (var writer = new StringWriter())
```

Comment:
Had to move away from passing the file stream directly. It started throwing "another process is using this file" exception. So delegating it to `ServerLogger` is best. Refer https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Razor.Tools/Roslyn/CompilerServerLogger.cs#L116